rpm/stackdriver-agent

```diff
@@ -244,6 +244,9 @@ case "$1" in
         exit "$rc"
         ;;
     condrestart|try-restart)
+        if [[ "$1" == "condrestart" ]]; then
+            log_warning_msg "Use try-restart rather than condrestart"
```

Comment:
I would revert changes to the init scripts. `condrestart` works on many platforms, and where it doesn't, you'll never get to this message, as the `service` executable would complain and kick you out first.

Reply:
Are there any that support `condrestart` and _not_ `try-restart`?  `condrestart` concerns me because calling it when the service is running results in `active (exited)` status, whereas `try-restart` results in `active (running)`. They're supposed to be consistent IIRC, but doing more for that fix is outside the scope of this change.

Reply:
We should support both: https://fedoraproject.org/wiki/EPEL:SysVInitScripts#condrestart_and_try-restart

Reply:
"must not differ in behavior" -- but they do...

Reply:
Not at the init script level — they literally resolve to the exact same snippet of code. The difference in behavior you saw was from a really old systemd (234, as far as I can see), and that's not our problem.

Reply:
/etc/init.d/stackdriver-agent condrestart --> active (exited)
/etc/init.d/stackdriver-agent try-restart --> active (running)

If that's all fine and dandy, I'll make the change

Reply:
It's definitely a systemd problem:
```
$ sudo /etc/init.d/stackdriver-agent try-restart
redirecting to systemctl try-restart stackdriver-agent.service
```
vs
```
$ sudo /etc/init.d/stackdriver-agent condrestart
Trying to restart Stackdriver metrics collection agent: stackdriver-agent (warning).
option = PIDFile; value = /var/run/stackdriver-agent.pid;
option = Interval; value = 60.000000;
option = Hostname; value = ;
option = FQDNLookup; value = false;
Created new plugin context.
.
```
Note that systemd recognizes and intercepts `try-restart`, but `condrestart` is handled directly by the script. When the process restarts outside of systemd, it confuses systemd, so it thinks the service is dead. This is all normal, and we should just accept that behavior. Newer versions of systemd will intercept `condrestart` as well.

So yes, this difference in behavior is fine.

Reply:
wooHooOooooO!